### PR TITLE
copy CI config from flasher

### PR DESCRIPTION
**Pull request by DoctorVin**

### CODE REVIEW

File: .github/workflows/push-pr-lint.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        with:
+          go-version: '1.20'
+
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: build binary
+        run: make build-linux
+
+      - name: Set up Docker Buildx
```
Comment (joelrebel) [inline .github/workflows/push-pr-lint.yaml:36]: all the docker/image bits can be excluded

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+jobs:
+  goreleaser:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      id-token: write
+      packages: write
+    steps:
+      -
+        name: Login to GHCR
```
Comment (joelrebel) [inline .github/workflows/release.yml:17]: same here, purge the GHCR login and docker action.

File: .goreleaser.yaml [hunk 3]
```diff
@@ -0,0 +1,86 @@
+project_name: flasher
```
Comment (joelrebel) [inline .goreleaser.yaml:1]: update project name

File: .goreleaser.yaml [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+builds:
+  - id: go
+    env:
+      - CGO_ENABLED=0
+    goos:
+      - linux
+      - darwin
+    ldflags:
+      - -X "github.com/metal-toolbox/flasher/internal/version.AppVersion={{ .Version }}"
```
Comment (joelrebel) [inline .goreleaser.yaml:14]: we don't have a version command in mctl yet, maybe we should add that in so users can verify the version thier running?

File: .goreleaser.yaml [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+      - "sign-blob"
+      - "--oidc-issuer=https://token.actions.githubusercontent.com"
+      - "--output-certificate=${certificate}"
+      - "--output-signature=${signature}"
+      - "${artifact}"
+      - "--yes" # required on cosign 2.0.0+
+    artifacts: all
+    output: true
+
+docker_signs:
```
Comment (joelrebel) [inline .goreleaser.yaml:78]: docker signs can go

**Review by joelrebel — COMMENTED** [review]
requires a bit more clean up

File: .github/workflows/push-pr-lint.yaml (hunk 1 above)
Comment (DoctorVin) [reply .github/workflows/push-pr-lint.yaml:36]: kept these is as per our discussion on slack

File: .goreleaser.yaml (hunk 4 above)
Comment (DoctorVin) [reply .goreleaser.yaml]: I added a version command now.

File: .github/workflows/release.yml (hunk 2 above)
Comment (DoctorVin) [reply .github/workflows/release.yml:17]: Done.

**Review by joelrebel — APPROVED** [review]

**Pull request merged**